rasgoql/rasgoql/primitives/enums.py

```diff
@@ -0,0 +1,116 @@
+"""
+Enums
+"""
+from enum import Enum
+
+from rasgoql.errors import ParameterException
+
+
+def _wrap_in_quotes(string: str) -> str:
+    return "'"+string+"'"
+
+class DWType(Enum):
+    """
+    Supported Data Warehouses
+    """
+    #FUTURE: BIGQUERY = 'bigquery'
+    #FUTURE: POSTGRES = 'postgres'
+    SNOWFLAKE = 'snowflake'
+
+SUPPORTED_DWS = _wrap_in_quotes("', '".join([e.value for e in DWType]))
+
+def check_data_warehouse(input_value: str):
```

Comment:
What's the added value of this check?  If they have to pass a valid `DataWarehouse` to the main RasgoQL, they shouldn't be able to fail this check, no? Or are you worried about some case where they try to create their own custom Credentials class?